packages/react-app/src/components/modals/TokenSelectorModal.jsx

```diff
@@ -49,26 +59,46 @@ export const TokenSelectorModal = ({ isOpen, onClose, onCustom }) => {
   const [tokenList, setTokenList] = useState([]);
   const [filteredTokenList, setFilteredTokenList] = useState([]);
   const smallScreen = useBreakpointValue({ sm: false, base: true });
-  const { getBridgeChainId, getGraphEndpoint } = useBridgeDirection();
+  const {
+    getBridgeChainId,
+    getGraphEndpoint,
+    enableForeignCurrencyBridge,
+  } = useBridgeDirection();
 
   // Callbacks
   const fetchTokenListWithBalance = useCallback(
     async tList => {
+      const tokenValueSortFn = ({ balance: balanceA }, { balance: balanceB }) =>
+        balanceB.sub(balanceA).gt(0) ? 1 : -1;
+
       const tokenListWithBalance = await Promise.all(
         tList.map(async token => ({
           ...token,
-          balance: await fetchTokenBalanceWithProvider(
-            ethersProvider,
-            token,
-            account,
-          ),
+          balance:
+            token.mode === 'NATIVE'
+              ? await ethersProvider.getBalance(account)
```

Comment:
@sunguru98 if we are adding that `mode === NATIVE` check inside `fetchTokenBalanceWithProvider`, then we can revert these changes to directly call `fetchTokenBalanceWithProvider` for all tokens?